sdk/lib/opentelemetry/sdk/trace/span.rb

```diff
@@ -373,9 +401,26 @@ def append_event(events, event) # rubocop:disable Metrics/AbcSize, Metrics/Cyclo
           events << event
         end
 
-        def wall_clock(timestamp)
-          timestamp = (timestamp.to_r * 1_000_000_000).to_i unless timestamp.nil?
-          timestamp || Process.clock_gettime(Process::CLOCK_REALTIME, :nanosecond)
+        def relative_timestamp(timestamp)
```

Comment:
🤔 I feel like we may be missing a concept here that encapsulates the time management strategy here. 

Perhaps objects like a `RelativeTime` vs a `RealTime` instead of making them functions on the span. 

I don't have an explicit ask here, just a thought experiment.

Reply:
They're all local to `SDK::Trace::Span` - we don't need a reusable abstraction, and I'm explicitly trying to avoid object allocation (the values here should all be machine words, except for explicit timestamps).